### PR TITLE
Implement packages filtering by section

### LINKED ISSUE #868: How about a filter for viewing packages
I couldn't find any other similar request/suggestion, please crosspost if there is. 

Would love to filter my changes-view. Right now it's bloated with a lot of themes. Would like to filter package type, e.g. Themes/Tweaks/Addons. Preferably enable or disable Zebra-wide in settings.

Perhaps the option to disable viewing all changes also?

### PR DESCRIPTION
**Pull request by PoomSmart**
Screenshot

Package filtering feature can be accessed from any section readout of any sources, changes are source-specific.

Known limitation: Changes to filtering requires refresh in Changes tab.

Closes #868

### CODE REVIEW

File: Zebra/Database/ZBDatabaseManager.h [hunk 1]
```diff
@@ -209,6 +215,20 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (NSURL *)paymentVendorURLForRepo:(ZBSource *)repo;
 
+/*!
+ @brief Determine whether a package section is enabled.
+ @param section The section name to determine.
+ @return A boolean value telling whether the package section is enabled.
+ */
+- (BOOL)isSectionEnabled:(NSString *)section;
```
Comment (wstyres) [inline Zebra/Database/ZBDatabaseManager.h]: Should this have a specifier for which `ZBSource` the section is hidden from? Some users might want to hide a section from Packix for example but want it to be visible for Dynastic.

File: Zebra/Database/ZBDatabaseManager.m [hunk 2]
```diff
@@ -735,12 +740,58 @@ - (int)numberOfPackagesInRepo:(ZBSource * _Nullable)repo section:(NSString * _Nu
     return NULL;
 }
 
+- (BOOL)isSectionEnabled:(NSString *)section {
+    if (section == nil) return NO;
+    if ([self openDatabase] == SQLITE_OK) {
+        NSString *filterQuery = [NSString stringWithFormat:@"SELECT ENABLED FROM PACKAGES_FILTER WHERE SECTION = \'%@\'", section];
```
Comment (wstyres) [inline Zebra/Database/ZBDatabaseManager.m]: Yeah looks like this query ignores the repo, I would add per-repo section hiding before I merge.

File: Zebra/Parsel/parsel.c [hunk 3]
```diff
@@ -96,6 +96,10 @@ char *updatesSchema() {
     return "UPDATES(PACKAGE STRING PRIMARY KEY, VERSION VARCHAR(16) NOT NULL, IGNORE INTEGER DEFAULT 0)";
 }
 
+char *packagesFilterSchema() {
+    return "PACKAGES_FILTER(SECTION STRING PRIMARY KEY, ENABLED BOOLEAN)";
```
Comment (wstyres) [inline Zebra/Parsel/parsel.c]: Maybe just add a repoID column here for per-repo?

File: Zebra/Database/ZBDatabaseManager.h (hunk 1 above)
Comment (PoomSmart) [reply Zebra/Database/ZBDatabaseManager.h]: Fixed.

File: Zebra/Database/ZBDatabaseManager.m (hunk 2 above)
Comment (PoomSmart) [reply Zebra/Database/ZBDatabaseManager.m]: Fixed.

File: Zebra/Parsel/parsel.c (hunk 3 above)
Comment (PoomSmart) [reply Zebra/Parsel/parsel.c]: Yes, fixed.

**Comment by PoomSmart** [comment]
Also fixed blank Changes page and checkbox at the All Packages row.

**Pull request merged**